### PR TITLE
Add Inside and Outside Motorspace Logs

### PR DESCRIPTION
**Pull request by Xav1204**
We needed to log in our csv file when the controller is Inside or Outside the motorspace to get better analysis of the games.

### CODE REVIEW

File: Assets/Scripts/Pointers/BubbleDisplay.cs [hunk 1]
```diff
@@ -172,8 +176,34 @@ public void InstantiateInCircle(GameObject container, int howMany, float radius)
...
+
+    private IEnumerator OutOfBoundAnimation(bool hoverBound, bool isOutOfBound)
+    {
+        if(hoverBound){
+            StartCoroutine(PointerHoverBoundaries(true));
+        }
+        
+        else{
+            StartCoroutine(PointerHoverBoundaries(false));
+        }
```
Comment (bastianilso) [inline Assets/Scripts/Pointers/BubbleDisplay.cs:206]: Hi @Xav1204 it is not clear to me why we would need to redefine the PointerHoverBoundaries function and make BubbleDisplay inherit Pointer, just to do some logging. 

Why not put the `loggerNotifier.NotifyLogger() ` statement inside the `if (render)` and `if (!render)`  inside the `Update()` function of BubbleDisplay?

https://github.com/med-material/Whack_A_Mole_VR/blob/master/Assets/Scripts/Pointers/BubbleDisplay.cs#L106

**Review by bastianilso — CHANGES_REQUESTED** [review]

File: Assets/Scripts/Pointers/BubbleDisplay.cs (hunk 1 above)
Comment (Xav1204) [reply Assets/Scripts/Pointers/BubbleDisplay.cs]: @bastianilso it's good, I put in the loggerNotifier inside the if conditions directly.

File: Assets/Scripts/Pointers/BubbleDisplay.cs [hunk 2]
```diff
@@ -116,6 +118,10 @@ void Update()
                 controllerRender.SetActive(true);
                 motorSpaceRender.color = motorActiveColor;
                 enterMotorStateEvent.Invoke(true);
+                loggerNotifier.NotifyLogger("Pointer Inside MotorSpace", EventLogger.EventType.PointerEvent, new Dictionary<string, object>()
+                {
+                    {"ControllerHoverBoundaries", controllerInsideBoundaries}
```
Comment (bastianilso) [inline Assets/Scripts/Pointers/BubbleDisplay.cs:123]: We won't need a dedicated column called `ControllerHoverBoundaries` in our log files. For now this event can just be logged with no additional columns, like exemplified below.

```
loggerNotifier.NotifyLogger("Pointer Inside MotorSpace", EventLogger.EventType.PointerEvent);
```

File: Assets/Scripts/Pointers/BubbleDisplay.cs [hunk 3]
```diff
@@ -80,6 +80,8 @@ public class EnterMotorSpaceEvent : UnityEvent<bool> {}
     public EnterMotorSpaceEvent enterMotorStateEvent;
     
     private bool render = true;
+    protected int controllerOutsideBoundaries = 0;
+    protected int controllerInsideBoundaries = 1;
```
Comment (bastianilso) [inline Assets/Scripts/Pointers/BubbleDisplay.cs:84]: These integers are only read and never set. I don't think we will need them, see my other comment.

File: Assets/Scripts/Pointers/Pointer.cs [hunk 4]
```diff
@@ -109,6 +109,7 @@ void Start()
             {"PointerShootOrder", "NULL"},
             {"ControllerName", "NULL"},
             {"ControllerAimAssistState", "NULL"},
+            {"ControllerHoverBoundaries", "NULL"},
```
Comment (bastianilso) [inline Assets/Scripts/Pointers/Pointer.cs:112]: Not neccessary, see other comment.

**Review by bastianilso — CHANGES_REQUESTED** [review]
Thanks for the changes @Xav1204, this is looking better, but I still think we can simplify this further, see code comments.

File: Assets/Scripts/Pointers/BubbleDisplay.cs [hunk 5]
```diff
@@ -4,7 +4,7 @@
 using UnityEngine.UI;
 using UnityEngine.Events;
 
-public class BubbleDisplay : MonoBehaviour
+public class BubbleDisplay : Pointer
```
Comment (bastianilso) [inline Assets/Scripts/Pointers/BubbleDisplay.cs:7]: unless we have a good reason to change the class inheritance, this should remain `public class BubbleDisplay : MonoBehaviour`.

**Review by bastianilso — CHANGES_REQUESTED** [review]
Code changes look much better and simpler now,  only one part left.

File: Assets/Scripts/Pointers/BubbleDisplay.cs (hunk 5 above)
Comment (Xav1204) [reply Assets/Scripts/Pointers/BubbleDisplay.cs]: @bastianilso It's good, I changed the class inheritance. I tested to see the difference between the private LoggerNotifier class instance and the class inheritance Pointer. I forgot to change it after my test. Sorry.

**Review by bastianilso — APPROVED** [review]
I think this looks good now.

**Pull request merged**